Docs/Architecture Decision Records/001 - PlainTextDeserializationVisitor.md

```diff
@@ -0,0 +1,22 @@
+# 1. Support "deserialization" of plain text 
+
+Date: 2020-05-07
+
+## Status
+
+Accepted
+
+## Context
+
+Some APIs, although they declare themselves as REST, return **a plain text string** as response. This breaks the serializer flow as it is expecting, by default, a JSON string. The result is a failure in deserialization and **the processor is interrupted**.
```

Comment:
REST is just a way to represent a state transfer https://en.wikipedia.org/wiki/Representational_state_transfer, and by the definition, can return multiple responses in different formats. The issue is how we manage that in framework. Also we are closing this to APIs, but for instance we could have the same case when consuming messages. Based on that I would try to put the context on a way like this
```
When connecting to external systems, those can return different format of responses like json, xml, yml, csv and so on. JMS supports the deserialization of few, but not all of them. This means that if the code tries to deserialize a response that is not in the expected format, the process is interrupted.
```
```

Reply:
I reworded the paragraph in a more abstract and generic way.